addons/mass_mailing/models/mass_mailing.py

```diff
@@ -563,14 +624,30 @@ def copy(self, default=None):
     def _group_expand_states(self, states, domain, order):
         return [key for key, val in type(self).state.selection]
 
-    def update_opt_out(self, email, res_ids, value):
-        model = self.env[self.mailing_model_real].with_context(active_test=False)
-        if 'opt_out' in model._fields:
-            email_fname = 'email_from'
-            if 'email' in model._fields:
-                email_fname = 'email'
-            records = model.search([('id', 'in', res_ids), (email_fname, 'ilike', email)])
-            records.write({'opt_out': value})
+    def update_opt_out(self, email, list_ids, value):
```

Comment:
This one can be rewriten in a simple way as we don't have opt-out fields anymore. We opt-out from mailing lists and that's about it.